File: Chapter05-old/Exercise5_10/Exercise5_10.py
spark = SparkSession \
   .builder \
   .appName("exercise_ten") \
   .getOrCreate()

from pyspark.sql.types import StructType, StructField, StringType, LongType

sales_schema = StructType([
  StructField("user_id", LongType(), False),
  StructField("product_item", StringType(), True)
])

sales_field = StructField("total_sales", LongType(), True)

another_schema = sales_schema.add(sales_field)

print(another_schema)

File: Chapter05-old/Exercise5_10/Exercise5_10.scala
val spark = SparkSession
   .builder()
   .appName("exercise_ten")
   .getOrCreate()

import org.apache.spark.sql.types.{StructType, StructField, StringType, LongType}

val sales_schema = StructType(List(
 StructField("user_id", LongType, true),
 StructField("product_item", StringType, true)
))

val sales_field = StructField("total_sales", LongType, true)

val another_schema = sales_schema.add(sales_field)

println(another_schema)
